server/public/model/utils.go

```diff
@@ -531,6 +531,12 @@ func ToJSON(v any) []byte {
 	return b
 }
 
+// FromJSON decodes an arbitrary object
+func ObjectFromJSON[T any](rd io.Reader, obj T, maxBytes int64) error {
```

Comment:
Nit: With no return value, the type parameter is not longer needed.
```suggestion
func ObjectFromJSON(rd io.Reader, obj any, maxBytes int64) error {
```